test/src/expectEvent.truffle.test.js

```diff
@@ -463,12 +523,161 @@ contract('expectEvent (truffle contracts)', function ([deployer]) {
             ));
           });
 
-          it('throws if an incorrect emitter is passed', async function () {
+          it('throws if an incorrect emitter class is passed', async function () {
             await assertFailure(expectEvent.inTransaction(this.txHash, EventEmitter, 'IndirectString',
               { value: this.value }
             ));
           });
+
+          it('throws if an incorrect emitter object is passed', async function () {
+            await assertFailure(
+              expectEvent.inTransaction(
+                this.txHash,
+                await IndirectEventEmitter.new(),
+                'IndirectString',
+                { value: this.value }
+              )
+            );
+          });
+        });
+      });
+    });
+
+    describe('with indexed event parameters', function () {
+      beforeEach(async function () {
+        this.indexedValue = new BN(42);
+        this.normalValue = new BN(2014);
+        this.indexedValue2 = new BN(2016);
+        this.normalValue2 = new BN(2020);
+        const { receipt } = await this.emitter.emitIndexedUintAndEmitIndirectly(
+          this.indexedValue,
+          this.normalValue,
+          this.indexedValue2,
+          this.normalValue2,
+          this.secondEmitter.address
+        );
+        this.txHash = receipt.transactionHash;
+      });
+
+      context('with directly called contract', function () {
+        it('accepts emitted events with correct indexed parameter and emitter object', async function () {
+          expectEvent.inTransaction(this.txHash, this.emitter, 'IndexedUint', {
```

Comment:
Here it is using n artifact with deployed address, don't throws at emitter.address